AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2664,6 +2668,67 @@ - (void)_enqueueAsyncSizingWithSentinel:(ASSentinel *)sentinel completion:(void(
   return self;
 }
 
+#pragma mark - ASEnvironment
+
+- (ASEnvironmentCollection *)environmentCollection
+{
+  return &_environmentCollection;
+}
+
+- (ASDisplayNode *)parent
+{
+  return self.supernode;
+}
+
+- (void)setParent:(ASDisplayNode *)parent
+{
+  [self __setSupernode:parent];
+}
+
+- (NSArray<ASDisplayNode *> *)children
+{
+  return self.subnodes;
+}
+
+- (BOOL)supportsMultipleChildren
```

Comment:
We should find a better name.  Until then, comment here that this really is intended to reference the fact that upward propogation of ASLayoutable values from ASLayoutSpec to ASDisplayNode is permitted - and that Nodes have "only one layoutspec child", even though clearly may have multiple subnodes.